packages/field-base/src/field-aria-controller.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * A controller to associate field slotted elements via ARIA attributes with a target element.
+ */
+export class FieldAriaController {
+  constructor(host) {
+    this.host = host;
+  }
+
+  setTarget(target) {
+    this.__target = target;
+    this.__setAriaRequiredAttribute(this.__required);
+    this.__setLabelIdToAriaAttribute(this.__labelId);
+    this.__setErrorIdToAriaAttribute(this.__errorId);
+    this.__setHelperIdToAriaAttribute(this.__helperId);
+  }
+
+  setRequired(required) {
+    this.__setAriaRequiredAttribute(required);
+    this.__required = required;
+  }
+
+  setLabelId(labelId) {
+    this.__setLabelIdToAriaAttribute(labelId, this.__labelId);
+    this.__labelId = labelId;
+  }
+
+  setErrorId(errorId) {
+    this.__setErrorIdToAriaAttribute(errorId, this.__errorId);
+    this.__errorId = errorId;
+  }
+
+  setHelperId(helperId) {
+    this.__setHelperIdToAriaAttribute(helperId, this.__helperId);
+    this.__helperId = helperId;
+  }
+
+  /** @private */
+  get __isGroupField() {
+    return this.__target === this.host;
```

Comment:
Would it be possible to delay intializing the controller until `setTarget` is called? Then we can set target in the constructor and avoid this comparison (or cache it by storing a reference to `this.__isGroupField` property).

I agree it might create unnecessary complication by having to wait for `slotchange` which is an async operation but with moving to Lit we could hopefully get this part working in a quite  straightforward way, e.g. like this:

```js
firstUpdated() {
  this._fieldAriaController = new FieldAriaController(this, this._inputNode);  
}
```

Reply:
I was thinking too about passing the target element as the second argument to the controller. However, I'm not sure what to do with `this._fieldAriaController.setHelperId`-like calls which might actually be invoked before the target element is set. Note, it is handled now, the ids are stored and applied once the target element is available. 
How is it supposed to work if the initialization is postponed until the target element is available?

Also, I'm not sure if I totally understand the idea to wait for a `slotchange` event, which element did you mean to subscribe to `slotchange`?

Reply:
Sorry, I messed up things with a `slotchange` event a bit. We don't really need to care about it.

My main point here: with the previous approach we were using `ariaTargetChanged` observer, but hopefully we can avoid that when using Lit. Let's not try to change it now, the current API seems good also.